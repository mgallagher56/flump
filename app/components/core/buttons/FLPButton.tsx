import React from 'react';

import type { ButtonProps } from '@chakra-ui/react';
import { Button } from '@chakra-ui/react';
import * as pressable from '@zag-js/pressable';
import type { PressEvent } from '@zag-js/pressable/dist/pressable.types';
import { normalizeProps, useMachine } from '@zag-js/react';

interface FLPButtonProps extends ButtonProps {
  preventFocusOnPress?: boolean;
  onPress?: (e: PressEvent) => void;
  onLongPress?: (e: PressEvent) => void;
}

const FLPButton: React.FC<React.PropsWithChildren<FLPButtonProps>> = ({
  children,
  colorScheme = 'blue',
  isDisabled,
  preventFocusOnPress,
  variant = 'solid',
  onLongPress,
  onPress,
  isLoading
}) => {
  const [state, send] = useMachine(
    pressable.machine({
      id: 'pressableBaseButton',
      onPress,
      onLongPress,
      disabled: isDisabled,
      preventFocusOnPress
    })
  );

  const api = pressable.connect(state, send, normalizeProps);

  return (
    <Button {...api.pressableProps} variant={variant} colorScheme={colorScheme} isLoading={isLoading}>
      {children}
    </Button>
  );
};

export default FLPButton;
