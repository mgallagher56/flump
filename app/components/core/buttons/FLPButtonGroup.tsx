import React from 'react';

import type { ButtonGroupProps } from '@chakra-ui/react';
import { ButtonGroup } from '@chakra-ui/react';

const FLPButtonGroup: React.FC<React.PropsWithChildren<ButtonGroupProps>> = (props) => {
  return <ButtonGroup {...props}>{props.children}</ButtonGroup>;
};

export default FLPButtonGroup;
