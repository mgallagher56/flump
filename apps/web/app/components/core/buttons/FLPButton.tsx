import { type FC, type PropsWithChildren } from 'react';

import type { ButtonProps } from '@chakra-ui/react';
import { Button } from '@chakra-ui/react';

interface FLPButtonProps extends ButtonProps {
  padding?: number;
}

const FLPButton: FC<PropsWithChildren<FLPButtonProps>> = ({
  colorScheme = 'blue',
  disabled,
  isLoading,
  padding,
  variant = 'solid',
  onClick,
  ...buttonProps
}) => {
  return (
    <Button
      colorScheme={colorScheme}
      isDisabled={disabled}
      isLoading={isLoading}
      padding={padding}
      variant={variant}
      onClick={onClick}
      width={'max-content'}
      {...buttonProps}
    >
      {buttonProps.children}
    </Button>
  );
};

export default FLPButton;
