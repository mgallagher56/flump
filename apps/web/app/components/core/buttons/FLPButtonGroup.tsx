import type { FC, PropsWithChildren } from 'react';

import { Group, type GroupProps } from '@chakra-ui/react';

const FLPButtonGroup: FC<PropsWithChildren<GroupProps>> = (props) => {
  return <Group {...props}>{props.children}</Group>;
};

export default FLPButtonGroup;
