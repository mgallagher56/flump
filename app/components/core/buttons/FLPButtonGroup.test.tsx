import { render } from '@testing-library/react';
import { describe, expect, test } from 'vitest';

import FLPButton from './FLPButton';
import FLPButtonGroup from './FLPButtonGroup';

describe('FLPButtonGroup', () => {
  test('renders as expected', () => {
    const { container, getByText } = render(
      <FLPButtonGroup>
        <FLPButton>Button 1</FLPButton>
        <FLPButton>Button 2</FLPButton>
      </FLPButtonGroup>
    );
    expect(container).toMatchSnapshot();
    expect(getByText('Button 1')).toBeDefined();
    expect(getByText('Button 2')).toBeDefined();
  });
});
