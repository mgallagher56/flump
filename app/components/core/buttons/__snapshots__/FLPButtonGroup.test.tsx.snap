// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FLPButtonGroup > renders as expected 1`] = `
<div>
  <div
    class="chakra-button__group css-1ind3v2"
    data-orientation="horizontal"
    role="group"
  >
    <button
      class="chakra-button css-1fysbuj"
      id="pressable:pressableBaseButton"
      type="button"
    >
      Button 1
    </button>
    <button
      class="chakra-button css-1fysbuj"
      id="pressable:pressableBaseButton"
      type="button"
    >
      Button 2
    </button>
  </div>
</div>
`;
